numba/core/ir.py

```diff
@@ -754,6 +754,35 @@ def get_targets(self):
         return []
 
 
+class DynamicRaise(Terminator):
+    """
+    Raise an exception class and some argument *values* unknown at compile-time.
+    Note that if *exc_class* is None, a bare "raise" statement is implied
+    (i.e. re-raise the current exception).
```

Comment:
What are the situations in which someone can reraise an exception? I see that code like this below won't work because we can't do `as e`
```python3
import numba

@numba.njit
def f(lst):
    try:
        idx = lst[10]
    except KeyError as e:
        raise e

f([1])
```